src/api/app/models/workflow/step/configure_repositories.rb

```diff
@@ -15,8 +17,10 @@ def call(_options = {})
     step_instructions[:repositories].each do |repository_instructions|
       repository = Repository.includes(:architectures).find_or_create_by(name: repository_instructions[:name], project: target_project)
```

Comment:
could `repository` be nil if the name is present but not valid, i.e `_MyRepo` ?

Reply:
The repository won't be `nil`, but it won't be persisted in the database. Here's an example:

```
#<Repository:0x000055c500b51df8 id: nil, db_project_id: 8205, name: "_lala", remote_project_name: "", rebuild: nil, block: nil, linkedbuild: nil, required_checks: []>
```

So the step wouldn't do anything with a repository with a non-valid name, since nothing would be persisted. I'll see what I can do about this. Informing the user should be enough.